backend/src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java

```diff
@@ -168,6 +168,14 @@ private Negotiation buildNegotiation(NegotiationCreateDTO negotiationBody, Reque
     negotiation.setResources(new HashSet<>(request.getResources()));
     negotiation.setHumanReadable(request.getHumanReadable());
     negotiation.setDiscoveryService(request.getDiscoveryService());
+
+    // Set default displayId if none provided
+    if (Objects.isNull(negotiation.getDisplayId()) || negotiation.getDisplayId().trim().isEmpty()) {
+      // Set default ID using pattern
+      String defaultId = "NEG-" + System.currentTimeMillis();
+      negotiation.setDisplayId(defaultId);
+    }
```

Comment:
are the comments necessary? Why not set the default id in the entity? How is this display Id more user friendly than the current one?

Reply:
1. Would leave the comments in, helps to understand some decisions later on :)
2. Agreed, will be set in the entity.
3. Display Id now at least contains just information about the creation date, not some UID. This should make it easier for users to identify the Negotiation by this ID.